src/Documenter.jl

```diff
@@ -305,9 +305,12 @@ function deploydocs(;
         # Change to the root directory and try to deploy the docs.
         cd(root) do
             Utilities.log("setting up target directory.")
-            Utilities.cleandir(target)
-            Utilities.log("building documentation.")
-            make()
+            isdir(target) || mkpath(target)
+            # Run extra build steps defined in `make` if required.
+            if make !== nothing
+                Utilities.log("building documentation.")
```

Comment:
Maybe have the logging be "running extra build steps." or something along those lines? "building documentation." could sound a bit like it's running `makedocs`.